lib/src/chart/line_chart/line_chart_painter.dart

```diff
@@ -1193,9 +1193,26 @@ class LineChartPainter extends AxisChartPainter<LineChartData> {
           mostTopOffset.dy - tooltipHeight - tooltipData.tooltipMargin;
     }
 
+    double tooltipLeft;
+    switch (tooltipData.tooltipAlignment) {
```

Comment:
Is it possible to move this logic into a function defined in their superclass (called `AxisChartPainter`)? 
In this way, we can re-use the logic (to respect the DRY pattern), and also we can write a unit-test for it. Because now this PR fails because there is no unit-test for this logic.

Reply:
I moved it, please check if this is what you meant.

Reply:
Yes, It is exactly what I mean.
I think now you need to write a unit test for it. Let's see the ci test-coverage result.
Thank you.